pkg/matcher/README.md

```diff
@@ -4,12 +4,18 @@
   * glob
   * regexp
   * simple patterns
+
+Depending on the symbol at the start of the string, the `matcher` will use one of the supported formats. 
+
+e.g `'* *'`: Because of the first `*`, `matcher` understands that it's in `glob` format, and according to the `glob` format the second `*` means that it will match *any* string. 
```

Comment:
I think this **specific** `e.g` shouldnt be there, at this point user is not aware of syntax so glob/not glob doesnt say much.

> Depending on the symbol at the start of the string, the `matcher` will use one of the supported formats

This is good because it doesnt contain specific details.

What we could do it to add a table like


| matcher           | short format  | long format           |
|-------------------|---------------|-----------------------|
| string            | `=`           |    `string`           |
| glob              | `*`           |    `glob`             |
| regexp            | `~`           |    `regexp`           |
| simple patterns   |               |    `simple_patterns`  |

Reply:
could be

| format          | description      |
|-------------------|---------------|
| `=`            | Match values that are exactly equal to the provided string           |
| `!=`            | Match values that are not equal to the provided string           |
| `*`            | Match values that glob-match the provided string           |
| `!*`            | Match values that do not glob-match the provided string           |
| `~`            | Match values that regexp-match the provided string           |
| `!~`            | Match values that do not regexp-match the provided string           |